src/emitter.hpp

```diff
@@ -34,31 +34,117 @@ using std::thread;
 using std::condition_variable;
 using std::mutex;
 
+/**
+ * @brief Emitter is responsible for sending events to a Snowplow Collector.
+ * 
+ * Once the emitter receives an event from the Tracker a few things start to happen:
+ * 
+ * 1. The event is added to a local Sqlite3 database (blocking execution)
+ * 2. A long running daemon thread is started which will continue to send events as long as they can be found in the database (asynchronous)
+ * 3. The emitter loop will grab a range of events from the database up until the SendLimit
+ * 4. The emitter will send all of these events as determined by the Request, Protocol and ByteLimits
+ *    - Each request is sent in its thread.
+ * 5. Once sent it will process the results of all the requests sent and will remove all successfully sent events from the database
+ */
 class Emitter {
 public:
+  /**
+   * @brief HTTP method used to send events to Snowplow Collector.
```

Comment:
Wondering if we need the `brief` tag for single line comment.
It's a honest question as I had the same doubts with the api docs for the mobile trackers but I've never found a satisfying answer.

Reply:
I know that the `brief` tag has a special meaning as the value will be displayed more prominently in docs and by IDEs. But I don't know what happens if it is missing, couldn't find the answer to that either...